.github/workflows/on-push.yml

```
@@ -0,0 +1,111 @@
+name: Continuous Integration
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  setup-js:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - uses: ./.github/actions/setup-node
+
+  # setup-rust:
+  #   runs-on: ubuntu-latest
+  #   steps:
+  #     - name: Checkout
+  #       uses: actions/checkout@v3
+  #     - uses: ./.github/actions/setup-rust
+
+  setup-python:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - uses: ./.github/actions/setup-python
+
+  # commitlint:
+  #   runs-on: ubuntu-latest
+  #   steps:
+  #     - name: Checkout
+  #       uses: actions/checkout@v3
+  #       with:
+  #         fetch-depth: 0
+  #     - name: Commit Lint
+  #       uses: wagoid/commitlint-github-action@v5.4.1
+  #       env:
+  #         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  eslint:
+    needs: setup-js
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup Node.js
+        uses: ./.github/actions/setup-node
+      - name: ESLint
+        run: yarn run lint:check
+
+  prettier:
+    needs: setup-js
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup Node.js
+        uses: ./.github/actions/setup-node
+      - name: Prettier
+        run: yarn run prettier:check
+
+  pylint:
+    needs: setup-python
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup Python
+        uses: ./.github/actions/setup-python
+      - name: PyLint
+        run: poetry run ruff check ./python
+
+  pytest:
+    needs: setup-python
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup Python
+        uses: ./.github/actions/setup-python
+      - name: PyTest
+        run: poetry run pytest
```

@sehyod @sergioramos Let's change this line to look only in the test directory for now. It shouldn't matter here, but if you run this line as is in our repo, pytest picks up a lot of other things that aren't truly tests.

```suggestion
        run: poetry run pytest python/tests
```

I tried to push a change up to test this, but I don't have `workflow` scope on the repo.

```
 ! [remote rejected] initial-ci-cd -> initial-ci-cd (refusing to allow a Personal Access Token to create or update workflow `.github/workflows/on-push.yml` without `workflow` scope)
error: failed to push some refs to 'https://github.com/refstudio/refstudio'
```